src/plugins/data/public/query/query_state.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Filter } from '@kbn/es-query';
+import type { TimefilterSetup } from './timefilter';
+import type { FilterManager } from './filter_manager';
+import type { QueryStringContract } from './query_string';
+import type { RefreshInterval, TimeRange, Query } from '../../common';
+
+/**
+ * All query state service state
+ */
+export interface QueryState {
+  time: TimeRange;
+  refreshInterval: RefreshInterval;
+  filters: Filter[];
+  query: Query;
+}
+
+export function getQueryState({
+  timefilter: { timefilter },
+  filterManager,
+  queryString,
+}: {
+  timefilter: TimefilterSetup;
+  filterManager: FilterManager;
+  queryString: QueryStringContract;
+}): QueryState {
+  return {
+    time: timefilter.getTime(),
```

Comment:
can we name this `timeRange`

Reply:
This would be problematic because `QueryState` is existing interface we use with URL syncing and `time` is how this key is persevered in URLs. 

I think to make it `timeRange` we will have to create a separate interface keep both around. WDYT?

Reply:
i think it makes sense to decouple the two, we'll be modifying this in the short future for universal search purposes and it sounds we'll be limited there if we keep the coupling

Reply:
but we can wait with that until we need to do some changes that would be incompatible, for the first step i think this should be fine. what do you think @flash1293

Reply:
Hmm, not sure why we want to decouple them - it seems like using the same interface for state syncing and communication between unified search and apps would be fine. Is there something I'm missing?

Reply:
The current `QueryState` interface reflects a state in the URL (partially _g and _a portion of it), so must be very careful with what we change there. 
We agreed with Peter, that would be safer to decouple and be explicit about what interface describes a state in the URL and what is for internal data transfer.

Reply:
Good point, didn't consider that. Also, no objections for this being called `time` for now.

Reply:
I went with a "partial" decoupling so in case the states state deviates, we will catch it on typescript level and can adjust then https://github.com/elastic/kibana/pull/132035/commits/30e9c665202fbcec3a198e32bf863da6e9361f77

I introduced `GlobalQueryStateFromUrl` state which for now matches `QueryState`. Everywhere where `QueryState` was used in the URL context I replaced it with `GlobalQueryStateFromUrl`